core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1943,24 +1944,43 @@ private RexNode convertOver(Blackboard bb, SqlNode node) {
 
     SqlNode windowOrRef = call.operand(1);
     final SqlWindow window =
-        validator.resolveWindow(windowOrRef, bb.scope, true);
+        validator.resolveWindow(windowOrRef, bb.scope);
 
-    // ROW_NUMBER() expects specific kind of framing.
-    if (aggCall.getKind() == SqlKind.ROW_NUMBER) {
-      window.setLowerBound(SqlWindow.createUnboundedPreceding(SqlParserPos.ZERO));
-      window.setUpperBound(SqlWindow.createCurrentRow(SqlParserPos.ZERO));
-      window.setRows(SqlLiteral.createBoolean(true, SqlParserPos.ZERO));
+    SqlNode sqlLowerBound = window.getLowerBound();
+    SqlNode sqlUpperBound = window.getUpperBound();
+    boolean rows = window.isRows();
+    SqlNodeList orderList = window.getOrderList();
+
+    if (!aggCall.getOperator().allowsFraming()) {
+      // If the operator does not allow framing, bracketing is implicitly
+      // everything up to the current row.
+      sqlLowerBound = SqlWindow.createUnboundedPreceding(SqlParserPos.ZERO);
+      sqlUpperBound = SqlWindow.createCurrentRow(SqlParserPos.ZERO);
+      if (aggCall.getKind() == SqlKind.ROW_NUMBER) {
+        // ROW_NUMBER() expects specific kind of framing.
+        rows = true;
+      }
+    } else if (orderList.size() == 0) {
+      // Without ORDER BY, there must be no bracketing.
+      sqlLowerBound = SqlWindow.createUnboundedPreceding(SqlParserPos.ZERO);
```

Comment:
Is this true ? From the MS-SQL syntax, the `ORDER BY` has no relationship with `<ROW or RANGE clause>`, they can both exist separately. [1] 

https://docs.microsoft.com/en-us/sql/t-sql/queries/select-over-clause-transact-sql?view=sql-server-ver15

Reply:
How can you say '5 ROWS PRECEDING' if you don't provide an ORDER BY clause to say by which criteria they are preceding?

If rows/range are present and there is no ORDER BY the validator doesn't throw (maybe it should) but we treat all rows as equal.

Reply:
You are right, i checked these for MS-SQL 2017
```sql
create table t(
  a int,
  b int
);

select max(a) over( rows 2 PRECEDING ) from t; -- syntax error throws
select max(a) over( order by b rows 2 PRECEDING ) from t; -- this works
```

Reply:
Show we throw for non-order by query ?

Reply:
I have logged https://issues.apache.org/jira/browse/CALCITE-3897. We should decide whether we want to fix it.